app/components/shared/inputs/BaseFormGroup.ts

```diff
@@ -7,17 +6,10 @@ import { BaseInput } from './BaseInput';
  * Base class for input-component layouts
  */
 export default abstract class BaseFormGroup extends BaseInput<any, IInputMetadata> {
-  @Prop()
-  readonly type: EInputType;
-
-  @Prop()
-  readonly value: undefined;
-
-  @Prop()
-  readonly metadata: IInputMetadata;
-
-  @Prop()
-  readonly title: string;
+  abstract readonly type: EInputType;
+  abstract readonly value: undefined;
+  abstract readonly metadata: IInputMetadata;
+  abstract readonly title: string;
```

Comment:
I realized that Vue doesn't like when you use `watchers` for fields that use  `@Prop` decorator in the base class.